app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -262,7 +265,11 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
             view = view
         )
 
-        binding.qrScanButton.visibility = if (context?.hasCamera() == true) View.VISIBLE else View.GONE
+        binding.qrScanButton.isVisible = when {
```

Comment:
```suggestion
        binding.qrScanButton.isVisible = !showUnifiedSearchFeature && context?.hasCamera() == true
```
Kinda weird we need to do `context?` here as well since we have to call `requireContext()` before this

Reply:
yes, this should just use `requireContext`

Reply:
I will avoid updating all of these though.  It can be a follow up to clean up.